src/catalog/database_catalog.cpp

```diff
@@ -1073,6 +1082,7 @@ bool DatabaseCatalog::SetClassPointer(transaction::TransactionContext *const txn
 
 bool DatabaseCatalog::SetIndexPointer(transaction::TransactionContext *const txn, const index_oid_t index,
                                       const storage::index::Index *const index_ptr) {
+  if (!TryLock(txn)) return false;
```

Comment:
This should probably be replaced by an assertion that the current transaction already owns the lock since this should only occur after a call to `CreateIndex`.